standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -13142,6 +13143,28 @@ public ReplicationMetricList getReplicationMetrics(GetReplicationMetricsRequest
     }
   }
 
+  @Override
+  public int deleteReplicationMetrics(int maxRetainSecs) {
+    if (maxRetainSecs < 0) {
+      LOG.debug("replication metrics deletion is disabled");
+      return 0;
+    }
+    boolean committed = false;
+    Query q = null;
+    try {
+      openTransaction();
+      long maxCreateTime = (int) (System.currentTimeMillis() / 1000) - maxRetainSecs;
```

Comment:
We are converting long to int earlier and now int to long. why not to use long everywhere. Don't we have chances of loosing the precision when System.currentTimeMillis becomes significantly higher than Integer.MAX_VALUE